shared/git/index.tsx

```diff
@@ -2,86 +2,97 @@ import * as React from 'react'
-  })
+// keep track in the module
+let moduleExpandedSet = new Set<string>()
 
-  renderItem = ({item}) => <Row key={item} {...this.rowPropsToProps(item)} />
+const Git = (props: Props) => {
+  const {error, loading, personals, teams, initialExpandedSet} = props
+  const {onShowDelete, onNewPersonalRepo, onNewTeamRepo} = props
 
-  renderSectionHeader = ({section}) => (
-    <Kb.SectionDivider label={section.title} showSpinner={section.loading} />
+  const [expandedSet, setExpandedSet] = React.useState(
+    new Set<string>(union(initialExpandedSet ?? new Set(), moduleExpandedSet))
   )
 
-  render() {
-    return (
-      <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true} style={styles.container}>
-        {!!this.props.error && <Kb.Banner color="red">{this.props.error.message}</Kb.Banner>}
-        {Styles.isMobile && (
-          <Kb.ClickableBox
-            ref={this.props.setAttachmentRef}
-            style={styles.header}
-            onClick={this.props.toggleShowingMenu}
-          >
-            <Kb.Icon
-              type="iconfont-new"
-              style={{marginRight: Styles.globalMargins.tiny}}
-              color={Styles.globalColors.blue}
-              fontSize={Styles.isMobile ? 20 : 16}
-            />
-            <Kb.Text type="BodyBigLink">New encrypted git repository...</Kb.Text>
-          </Kb.ClickableBox>
-        )}
-        <Kb.SectionList
-          keyExtractor={item => (typeof item === 'string' ? item : item.title)}
-          extraData={this.props.expandedSet}
-          renderItem={this.renderItem}
-          renderSectionHeader={this.renderSectionHeader}
-          sections={[
-            {data: this.props.personals, loading: this.props.loading, title: 'Personal'},
-            {data: this.props.teams, loading: this.props.loading, title: 'Team'},
-          ]}
-        />
-        <Kb.FloatingMenu
-          attachTo={this.props.getAttachmentRef}
-          closeOnSelect={true}
-          items={this.menuItems}
-          onHidden={this.props.toggleShowingMenu}
-          visible={this.props.showingMenu}
-          position="bottom center"
-        />
-      </Kb.Box2>
-    )
+  React.useEffect(() => {
+    moduleExpandedSet = expandedSet
+  }, [expandedSet])
+
+  const toggleExpand = (id: string) => {
+    moduleExpandedSet.has(id) ? moduleExpandedSet.delete(id) : moduleExpandedSet.add(id)
+    setExpandedSet(new Set(moduleExpandedSet))
   }
+
+  const menuItems = [
+    {icon: 'iconfont-person', onClick: onNewPersonalRepo, title: 'New personal repository'} as const,
+    {icon: 'iconfont-people', onClick: onNewTeamRepo, title: 'New team repository'} as const,
+  ]
+
+  const popupAnchor = React.useRef(null)
+  const {showingPopup, setShowingPopup, popup} = Kb.usePopup(popupAnchor, () => (
+    <Kb.FloatingMenu
+      attachTo={() => popupAnchor.current}
+      closeOnSelect={true}
+      items={menuItems}
+      onHidden={() => setShowingPopup(false)}
+      visible={showingPopup}
```

Comment:
If we always pass in `true` here, when we are not showing the whole menu just gets unmounted anyway because the hook returns `null` as `popup`. So is there a reason for `showingPopup` here to be returned by the hook?

Reply:
usePopup is generic, in cases where we're using floating menu (etc) you wouldn't need all the params but you might in other cases